functionalities/gdb_remote_client/TestqOffsets.py test fails on AArch64

The [TestqOffsets.py](https://github.com/llvm/llvm-project/blob/main/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py) test is failing on AArch64.  I even tried to test with d4eca120ac0af2a805c19301412bf843a71c14b5, which was the commit that added the test and test still fails, so it looks like it's never been working on the aarch64 system I'm using.

Output from the failed test:
```
FAIL: lldb-api :: functionalities/gdb_remote_client/TestqOffsets.py (522 of 2175)
******************** TEST 'lldb-api :: functionalities/gdb_remote_client/TestqOffsets.py' FAILED ********************
Script:
--
/usr/bin/python3 /home/gha/llvm-project/lldb/test/API/dotest.py -u CXXFLAGS -u CFLAGS --env LLVM_LIBS_DIR=/home/gha/llvm-project/build/./lib --env LLVM_INCLUDE_DIR=/home/gha/llvm-project/build/include --env LLVM_TOOLS_DIR=/home/gha/llvm-project/build/./bin --arch aarch64 --build-dir /home/gha/llvm-project/build/lldb-test-build.noindex --lldb-module-cache-dir /home/gha/llvm-project/build/lldb-test-build.noindex/module-cache-lldb/lldb-api --clang-module-cache-dir /home/gha/llvm-project/build/lldb-test-build.noindex/module-cache-clang/lldb-api --executable /home/gha/llvm-project/build/./bin/lldb --compiler /home/gha/llvm-project/build/./bin/clang --dsymutil /home/gha/llvm-project/build/./bin/dsymutil --make /usr/bin/gmake --llvm-tools-dir /home/gha/llvm-project/build/./bin --lldb-obj-root /home/gha/llvm-project/build/tools/lldb --lldb-libs-dir /home/gha/llvm-project/build/./lib --cmake-build-type Release /home/gha/llvm-project/lldb/test/API/functionalities/gdb_remote_client -p TestqOffsets.py
--
Exit Code: 1

Command Output (stdout):
--
lldb version 22.0.0git (https://github.com/llvm/llvm-project revision 0e3319537972d284e94a183899f15b40a798fd1f)
  clang revision 0e3319537972d284e94a183899f15b40a798fd1f
  llvm revision 0e3319537972d284e94a183899f15b40a798fd1f
Skipping the following test categories: ['libc++', 'msvcstl', 'dsym', 'gmodules', 'debugserver', 'objc']

--
Command Output (stderr):
--
FAIL: LLDB (/home/gha/llvm-project/build/bin/clang-aarch64) :: test (TestqOffsets.TestqOffsets.test)
======================================================================
FAIL: test (TestqOffsets.TestqOffsets.test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gha/llvm-project/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py", line 20, in test
    self.assertEqual(text.GetLoadAddress(target), 0x471000)
AssertionError: 18446744073709551615 != 4657152
Config=aarch64-/home/gha/llvm-project/build/bin/clang
----------------------------------------------------------------------
Ran 1 test in 0.028s

FAILED (failures=1)

```

--- a/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py
+++ b/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py
@@ -10,13 +10,10 @@ class TestqOffsets(GDBRemoteTestBase):
         def qOffsets(self):
             return "Text=470000;Data=470000"
 
-        def qfThreadInfo(self):
-            # Prevent LLDB defaulting to PID of 1 and looking up some other
-            # process when on an AArch64 host.
-            return "m-1"
-
     def test(self):
         self.server.responder = TestqOffsets.Responder()
+        # This ensures that we do not pick up any binaries on the host.
+        self.runCmd("platform select remote-linux")
         target = self.createTarget("qOffsets.yaml")
         text = target.modules[0].FindSection(".text")
         self.assertEqual(text.GetLoadAddress(target), lldb.LLDB_INVALID_ADDRESS)

--- a/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py
+++ b/lldb/test/API/functionalities/gdb_remote_client/TestqOffsets.py
@@ -10,6 +10,11 @@ class TestqOffsets(GDBRemoteTestBase):
         def qOffsets(self):
             return "Text=470000;Data=470000"
 
+        def qfThreadInfo(self):
+            # Prevent LLDB defaulting to PID of 1 and looking up some other
+            # process when on an AArch64 host.
+            return "m-1"
+
     def test(self):
         self.server.responder = TestqOffsets.Responder()
         target = self.createTarget("qOffsets.yaml")

